[Chore] Fix must-gather test

IshwarKanse opened the pull request:
**Description:** <Describe what has changed.>
The PR fixes must-gather test.

_Assisted by Claude using Gemini API._

IshwarKanse commented:
@frzifus We use AI to update the docs whenever a test changes. I’ve verified with Claude Sonnet 4 and Gemini—they can analyze the existing README structure and update it accordingly. These docs serve as blueprints to help users easily deploy and configure their observability stack. We don’t expect users/customers to rely on AI to understand test implementations.

atoulme commented:
Sorry, this PR is all over the place. I cannot follow it. Please break it in smaller chunks.

IshwarKanse commented:
@atoulme I removed the CLAUDE.md file, will add it in another PR. The rest of the changes are for the must-gather test, switched from using sidecar to deployment mode for collector.
